File: NutriTechBackOffice/ClientApp/src/app/interfaces/plan-alimentacion.ts
import { LastUpdated } from "./user";

export interface PlanAlimentacion {
  nombre: string;
  tipo: string;
  cantAguaDiaria: Number;
  cantColacionesDiarias: Number;
  desayuno: Array<string>;
  almuerzo: Array<string>;
  merienda: Array<string>;
  cena: Array<string>;
  colacion: Array<string>;
  lastUpdated: LastUpdated
}

File: NutriTechBackOffice.Domain/Entities/PlanAlimentacion.cs
using Google.Cloud.Firestore;
using System;
using System.Collections.Generic;
using System.Text;

namespace NutriTechBackOffice.Domain.Entities
{
    [FirestoreData]
    public class PlanAlimentacion
    {
        [FirestoreProperty]
        public string Nombre { get; set; }

        [FirestoreProperty]
        public string Tipo { get; set; }

        [FirestoreProperty]
        public float CantAguaDiaria { get; set; }

        [FirestoreProperty]
        public float CantColacionesDiarias { get; set; }

        [FirestoreProperty]
        public List<string> Desayuno { get; set; }

        [FirestoreProperty]
        public List<string> Almuerzo { get; set; }

        [FirestoreProperty]
        public List<string> Merienda { get; set; }

        [FirestoreProperty]
        public List<string> Cena { get; set; }

        [FirestoreProperty]
        public List<string> Colacion { get; set; }

        [FirestoreProperty, ServerTimestamp]
        public object LastUpdated { get; set; }
    }
}
